src/Microsoft.AspNetCore.Authentication.AzureAd/AzureAdB2COptions.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.AspNetCore.Authentication.AzureAd
+{
+    public class AzureAdB2COptions : OpenIdConnectOptions
+    {
+        public AzureAdB2COptions()
+        {
+            TokenValidationParameters = new TokenValidationParameters() { NameClaimType = "name" };
+
+            Events = new OpenIdConnectEvents()
+            {
+                OnRemoteFailure = context =>
+                {
+                    context.HandleResponse();
+                    // Handle the error code that Azure AD B2C throws when trying to reset a password from the login page 
+                    // because password reset is not supported by a "sign-up or sign-in policy"
+                    if (context.Failure is OpenIdConnectProtocolException && context.Failure.Message.Contains("AADB2C90118"))
+                    {
+                        // If the user clicked the reset password link, redirect to the reset password route
+                        context.Response.Redirect("/Account/ResetPassword");
+                    }
+                    else if (context.Failure is OpenIdConnectProtocolException && context.Failure.Message.Contains("access_denied"))
+                    {
+                        context.Response.Redirect("/");
```

Comment:
@danroth27 the `access_denied` error is totally standard so this could definitely live in the OIDC/OAuth2 middleware: https://github.com/aspnet/Security/issues/710.